add more detail (#11)
LisaFC/docsysite: content/en/docs/Updating/_index.md

 Keeping the theme up to date.
---

We hope to continue to make improvements to the theme [along with the Docsy community](/docs/contribution-guidelines/). 
If you have cloned the example site (or are otherwise using the theme as a submodule), you can update the Docsy theme
yourself. 

Updating Docsy means that your site will build using the latest version of Docsy at `HEAD` and include 
all the new commits or changes that have been merged since the time when you first added the Docsy submodule, 

> or last updated. If you have any modified content like a Docsy [customiztion](/docs/adding-content/lookandfeel/), those 

Reviewer: This won't affect any modifications you may have made to [override the Docsy look and feel](/docs/adding-content/lookandfeel) in your own project, as these don't modify the theme itself. For details of what has changed...

Reviewer: (just to make it clear that the modifications in question are overrides in your project, rather than changes to the theme itself)

Author: Thanks @LisaFC! FYI, if you click 
![image](https://user-images.githubusercontent.com/11762685/58900258-af4f7980-86b3-11e9-9cc1-5f52aa0f90a1.png) in the comment box, you can suggest an inline edit.

changes will remain in your project. For details about what has changed, see the list of 
[Docsy commits](https://github.com/google/docsy/commits/master).

> ## Update a Docsy clone

Reviewer: Maybe move this below the "updating a submodule", as we're recommending that people do the submodule version for easy updates, or at least updates that won't potentially involve weep-inducing merge madness.

If you have a local clone of the Docsy theme, you `git pull` all the new changes. For any modifications that you
made to the theme, you must manually resolve any merge conflicts. 

> For example, if `upstream`

Reviewer: Phrasing is a little awkward. Consider:

For example, here's how you do this if you've set `https://github.com/google/docsy.git` as your `upstream` remote. From the root directory where you cloned the Docsy theme, run:

Author: That cleaner, thanks. I guess i was over complicating it by indirectly suggesting that "upstream" could be named anything too (ie `docsy` could be your remote's name - `git pull docsy master`)

is your `remote` that is set to `https://github.com/google/docsy.git`, then from the root 
directory where you cloned the Docsy theme, you run:

```
git pull upstream master
```

## Update a Docsy submodule

If you are using the Docsy theme as a submodule in your project, you update the submodule:

1. Navigate to the root of your local project, then run:

    ```
    $ git submodule update --remote
    ```
    
1. Add and then commit the change to your project:

    ```
    $ git add themes/
    $ git commit -m "Updating theme submodule"
    ```

> 1. Push the commit to your repo. For example, if your `remote` is `origin`, then you run: 

Reviewer: The "if your remote is origin" bit seems a bit odd, if you've made a local copy of your repo and you're doing this work in it, the relevant remote is "origin" by default, you'd have to have explicitly set it to something else.

How about:
Push the commit to your project repo (by default this is your `origin` remote if you made a local copy of the repo):

Author: Per our chat, let me know if `origin` is set by default since im not getting  that behavior. I also intentionally left this vague since someone could have set their remote name as `docsy` or any other unique name.

Reviewer: Yes, it is - I just tried cloning the Kubeflow website (which I don't own) into a new directory on my Mac and origin is set to the Kubeflow website.

Author: Sorry! And thanks for trying this out for me! I tried again too and found that the correct command to list the remote and url is `git remote get-url origin` (where `git remote` only list the remote without the URL, but I incorrectly read that as "`origin` has no URL set"). 

```suggestion
1. Push the commit to your project repo (by default this is your `origin` remote):
```

If you have cloned the example site (or are otherwise using the theme as a submodule), you can update the Docsy theme
yourself. 

Updating Docsy means that your site will build using the latest version of Docsy at `HEAD` and include 
all the new commits or changes that have been merged since the point in time that you initially added the Docsy 
submodule, or last updated. Updating won't affect any modifications that you made in your own project to 
[override the Docsy look and feel](/docs/adding-content/lookandfeel/), as your overrides 
don't modify the theme itself. For details about what has changed in the theme, see the list of 
[Docsy commits](https://github.com/google/docsy/commits/master).

Depending on how you chose to use Docsy, follow the corresponding steps to update the theme:

## Update a Docsy submodule

If you are using the Docsy theme as a submodule in your project, you update the submodule:

1. Navigate to the root of your local project, then run:

    ```
    $ git submodule update --remote
    ```
    
1. Add and then commit the change to your project:

    ```
    $ git add themes/
    $ git commit -m "Updating theme submodule"
    ```

> 1. Push the commit to your project repo (by default this is your `origin` remote):

Reviewer: you really don't need to say that this is your origin remote as it's either a)automatically set to origin because you've just made a local copy of your own repo or b)we've explicitly told them to set it to origin if they've copied our example using the command line. It would be extremely weird for it to be anything else.

Author: removed
```suggestion
1. Push the commit to your project repo. For example, run:
```

    ```
    $ git push origin master
    ```

## Update a Docsy clone

If you have a local clone of the Docsy theme, you `git pull` all the new changes. For any modifications that you
made in the cloned theme, you must manually resolve any merge conflicts. 

> For example, if you've set https://github.com/google/docsy.git as your `upstream` remote, you run the following

Reviewer: ```suggestion
To do this, go to your theme root directory and run:

    git pull origin master

If you have made any local changes to the cloned theme, you must manually resolve any merge conflicts.
```

> For example, if you've set https://github.com/google/docsy.git as your `upstream` remote, you run the following

Author: Sorry, too used to the developer flow (first clone and then set upstream). I wonder if others might instinctively confuse these flows? Maybe our "Option 1" install instructions should instead refer to this as "Copy Docsy locally" (and then articulate that you use "git clone" and 'origin' should point to google/docsy)?

Just a personal note (which might resonate with others): This is also why I also specifically name all my remotes (or ensure that i always use "upstream" to point to a "not-mine" repo, and "origin" to my corresponding "fork") otherwise of the 10 cloned repos I have, knowing where "origin" points is no longer a guessing game (or a process where I have to run 'git remote -v' each time to remind myself)?

> For example, if you've set https://github.com/google/docsy.git as your `upstream` remote, you run the following

Reviewer: Hmm, I think copying the theme itself locally (versus following our recommended flows of either copying the entire example site via GUI/command line or adding the theme as a submodule) is something that only a fairly comfortable-with-Git user would do and I suspect they're not that likely to get confused by doing pulls from the theme master repo. The install flow this is the update flow for is this one: https://serene-roentgen-10fbaa.netlify.com/docs/getting-started/#cloning-the-docsy-theme-to-your-projects-themes-subdirectory. We could maybe link that from the "a local clone of the Docsy theme" in the intro sentence to make sure users don't think they need to do this if they did anything else?

I think for a lot of Git users naming "origin" and "upstream" as anything other than "origin" and "upstream" is going to be more rather than less confusing (it is for me...). The only potential difficulty here is that you've cloned the theme into a subdir of your repo so once you're in that subdir it has its own Git history/origin, but again, we're aiming this update process at people who've followed our "if you REEEEEALLLY want to do this you can but it's going to be harder to update" install process... (also we've given them instructions where if they just leave the default remote names, they'll work)

> For example, if you've set https://github.com/google/docsy.git as your `upstream` remote, you run the following

Author: Just brain storming since I didn't even recognize that I confused this flow until you called it out. I'm also planning on updating these docs to use the new "repo template" since that works so well for the flow for copying the docsy-example (to build from or play with) and wanted your input before I take a stab at that. We can keep 'origin' since it is the default ( and i'm likely the only confused person here ;p ).
